tests/e2e/nvidia-container-cli_test.go

```diff
@@ -0,0 +1,242 @@
+install -m 0755 -d /etc/apt/keyrings
+curl -fsSL https://download.docker.com/linux/ubuntu/gpg -o /etc/apt/keyrings/docker.asc
+chmod a+r /etc/apt/keyrings/docker.asc
+
+# Add the repository to Apt sources:
+echo "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.asc] https://download.docker.com/linux/ubuntu $(. /etc/os-release && echo \"${UBUNTU_CODENAME:-$VERSION_CODENAME}\") stable" | tee /etc/apt/sources.list.d/docker.list > /dev/null
+apt-get update
+
+apt-get install -y docker-ce docker-ce-cli containerd.io
+
+# start dockerd in the background
+dockerd &
+
+# wait for dockerd to be ready with timeout
+timeout=30
+elapsed=0
+while ! docker info > /dev/null 2>&1 && [ $elapsed -lt $timeout ]; do
+    echo "Waiting for dockerd to be ready..."
+    sleep 1
+    elapsed=$((elapsed + 1))
+done
+if [ $elapsed -ge $timeout ]; then
+    echo "Docker failed to start within $timeout seconds"
+    exit 1
+fi
+'`
+	installCTKTemplate = `docker exec -u root {{.ContainerName}} bash -c '
+# Create a temporary directory and rootfs path
+TMPDIR="$(mktemp -d)"
+
+# Expose TMPDIR for the child namespace
+export TMPDIR
+
+docker run --rm -v ${TMPDIR}:/host-tmpdir --entrypoint="sh" {{.ToolkitImage}}-packaging -c "cp -p -R /artifacts/* /host-tmpdir/"
+dpkg -i ${TMPDIR}/packages/ubuntu18.04/amd64/libnvidia-container1_*_amd64.deb ${TMPDIR}/packages/ubuntu18.04/amd64/nvidia-container-toolkit-base_*_amd64.deb ${TMPDIR}/packages/ubuntu18.04/amd64/libnvidia-container-tools_*_amd64.deb
+
+nvidia-container-cli --version
+'`
+
+	libnvidiaContainerCliTestTemplate = `docker exec -u root {{.ContainerName}} bash -c '
+# Create a temporary directory and rootfs path
+TMPDIR="$(mktemp -d)"
+ROOTFS="${TMPDIR}/rootfs"
+mkdir -p "${ROOTFS}"
+
+# Expose ROOTFS for the child namespace
+export ROOTFS TMPDIR
+
+# Download Ubuntu base image with error handling
+curl -fsSL http://cdimage.ubuntu.com/ubuntu-base/releases/22.04/release/ubuntu-base-22.04-base-amd64.tar.gz | tar -C $ROOTFS -xz || {
+    echo "Failed to download or extract Ubuntu base image"
+    exit 1
+}
+
+# Enter a new mount + PID namespace so we can pivot_root without touching the
+# container'\''s original filesystem.
+unshare --mount --pid --fork --propagation private -- sh -eux <<'\''IN_NS'\''
+  : "${ROOTFS:?}"
+
+  # 1 Bind-mount the new root and make the mount private
+  mount --bind "$ROOTFS" "$ROOTFS"
+  mount --make-private "$ROOTFS"
+  cd "$ROOTFS"
+
+  # 2 Minimal virtual filesystems
+  mount -t proc  proc proc
+  mount -t sysfs sys  sys
+  mount -t tmpfs tmp  tmp
+  mount -t tmpfs run  run
+
+  # 3 Configure NVIDIA devices
+  nvidia-container-cli --load-kmods configure --ldconfig=@/sbin/ldconfig.real --no-cgroups --utility --device 0 $(pwd)
+
+  # 4 Switch root into the prepared filesystem
+  pivot_root . mnt
+  umount -l mnt
+  nvidia-smi -L
+
+IN_NS
+'`
+
+	dockerRunCmdTemplate = `docker run -d --name {{.ContainerName}} --privileged --runtime=nvidia \
```

Comment:
I also don't think this needs to be a template.

Reply:
Done